app/backend/src/couchers/models.py

```diff
@@ -653,3 +653,262 @@ class InitiatedUpload(Base):
     @hybrid_property
     def is_valid(self):
         return (self.created <= func.now()) & (self.expiry >= func.now())
+
+
+class Node(Base):
+    """
+    Node, i.e. geographical subdivision of the world
+    """
+
+    __tablename__ = "nodes"
+
+    id = Column(BigInteger, primary_key=True)
+
+    name = Column(String, nullable=False)
+    parent_node = Column(BigInteger, ForeignKey("nodes.id"))
+    polygon = Column(Geometry(geometry_type="POLYGON", srid=4326))
+    official_cluster_id = Column(BigInteger, ForeignKey("clusters.id"), nullable=False, unique=True)
+    created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+
+
+class Cluster(Base):
+    """
+    Cluster, administered grouping of content
+    """
+
+    __tablename__ = "clusters"
+
+    id = Column(BigInteger, primary_key=True)
+    name = Column(String, nullable=False)
+    main_page_id = Column(BigInteger, ForeignKey("pages.id"), nullable=False, unique=True)
+    created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+
+
+class NodeClusterAssociation(Base):
+    """
+    NodeClusterAssociation, grouping of nodes
+    """
+
+    __tablename__ = "node_cluster_associations"
+    __table_args__ = (UniqueConstraint("node_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    node_id = Column(BigInteger, ForeignKey("nodes.id"), nullable=False)
+    cluster_id = Column(BigInteger, ForeignKey("clusters.id"), nullable=False)
+
+
+class GroupRole(enum.Enum):
+    member = 1
+    admin = 2
+
+
+class ClusterSubscription(Base):
+    """
+    ClusterSubscription of a user
+    """
+
+    __tablename__ = "cluster_subscriptions"
+    __table_args__ = (UniqueConstraint("user_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    user_id = Column(BigInteger, ForeignKey("users.id"), nullable=False)
+    cluster_id = Column(BigInteger, ForeignKey("clusters.id"), nullable=False)
```

Comment:
You should also add the `relationship`/`backref` stuff for foreign keys, so that we can do something like `ClusterSubscription.user`, see the models above.